src/core/qgsabstractcontentcache.h

```diff
@@ -367,14 +369,31 @@ class CORE_EXPORT QgsAbstractContentCache : public QgsAbstractContentCacheBase
         QMetaObject::invokeMethod( const_cast< QgsAbstractContentCacheBase * >( qobject_cast< const QgsAbstractContentCacheBase * >( this ) ), "onRemoteContentFetched", Qt::QueuedConnection, Q_ARG( QString, path ), Q_ARG( bool, true ) );
       } );
 
-      QgsApplication::taskManager()->addTask( task );
+      if ( synchrone )
+      {
+        if ( task->run() )
+        {
+          if ( mRemoteContentCache.contains( path ) )
+          {
+            // We got the file!
+            return *mRemoteContentCache[ path ];
+          }
+        }
+      }
+      else
+      {
+        QgsApplication::taskManager()->addTask( task );
+      }
       return fetchingContent;
     }
 
     void onRemoteContentFetched( const QString &url, bool success ) override
     {
       QMutexLocker locker( &mMutex );
-      mPendingRemoteUrls.remove( url );
+      if ( mPendingRemoteUrls.contains( url ) )
```

Comment:
This shouldn't be needed -- `remove` will happily handle the case when the value isn't present

Reply:
I was thinking that `remove` uncontained data made the segfault. I do not know this point.